### PR TITLE
New time locks

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Changes and add time-lock times on some jobs. Here is a recap of every job and their time-lock

// No Time lock
Marine
Engineer
Corpsman
SG
ST
MO
CL
Rsr

//1 hour
RO

//3 hours
SL
AI
PO
SO

//10 hours
FC
CSE
CMO
Synth

//15 hours
Captain

## Why It's Good For The Game

These hours are not much too ask if you compare to the griefing power these roles have. I don't know why or how you would play a role shipside where you can fuck the whole round if you don't have at least a basic understanding of the game ground side. 
You should not be in a position of leader if you can't tie your own laces

Having captain fresh from CM being here just to meme around is not that rare, and this PR can hopefully curb this a bit.

## Changelog
:cl:
admin: Stricter time lock for shipside and command jobs
/:cl:

### CODE REVIEW

**Comment by Chaznoodles** [comment]
When I said ten hours was the maximum I'd recommend, I didn't mean put all of the roles at ten hours...

**Comment by revelation-8468** [comment]
Ten hours to play anything other than a point and click simulator seems a bit harsh.

ANY timelock for shipside jobs seems a bit harsh, actually. For ST especially, as it's a job that people would probably use for LEARNING how to engineer in the first place, in an environment where they're not going to be screamed at and discouraged from playing further.

Just my thought, though.

**Comment by BraveMole** [comment]
> ANY timelock for shipside jobs seems a bit harsh, actually. For ST especially, as it's a job that people would probably use for LEARNING how to engineer in the first place, in an environment where they're not going to be screamed at and discouraged from playing further.

Hum, not a bad points, I've put ST and MO on the one hour mark

**Comment by TiviPlus** [comment]
Pending leads/admins OKing this

**Comment by Tennessee116** [comment]
Corporate Liaison is a **soft-power** job. Pardon me if I am wrong, but there is next to no responsibility and no actual power outside of Adminbus-backed RPing that they have.

Why would they need a time-lock of 10 hours?

**Comment by BraveMole** [comment]
Cl has no responsability, and very limited griefing powers. But, since it's the only heavy rp role we got, don't think it's a good idea for new players to play it. You are coming to TGMC mostly for action, it is then in my humble opinion better for them to propose the "traditional" roles of TGMC before quirky role like CL.

If you are a new player, and you can only chose between marine and CL, there's a chance you'll chose CL, get bored and regret your choice. You should play CL when you have a clear-ish idea of what funny RP you gonna do this round, else it's gonna be very boring.

That being said, i'm not ultra conviced by my own argument and I don't think it's a big deal if CL were put under 0,1 or 3 hours time lock

**Comment by Tennessee116** [comment]
> Cl has no responsability, and very limited griefing powers. But, since it's the only heavy rp role we got, don't think it's a good idea for new players to play it. 

The medical researcher is an equally roleplay-oriented role, and it has some extra optional responsibilities, considering that they can choose to partake in regular doctor activities. Speaking of which, why do _they_ have a timelock? It seems like a rather excellent medical entry job, considering the timelocking of medics.

> You are coming to TGMC mostly for action, it is then in my humble opinion better for them to propose the "traditional" roles of TGMC before quirky role like CL.

They are coming to TGMC for action, so we lock them from non-action roles? How does this even correlate? If they are coming for action then they will go for Squad [something] or xenomorph, hell, or survivor. 

 
> If you are a new player, and you can only chose between marine and CL, there's a chance you'll chose CL, get bored and regret your choice. You should play CL when you have a clear-ish idea of what funny RP you gonna do this round, else it's gonna be very boring.

If they can't figure out that marine is the combat role, how the _fuck_ do they know this is an action-oriented server?

**Comment by Drevenshek** [comment]
We already have a lack of support players on this server, I was mentor helping a new fella today who was playing as doctor and doing a fine job of it and he played that because his ping doesent let him go pewpew xenos. This kind of gatekeeping will really bottleneck us in the future. Lots of players come with experience roles form other servers that can easily translate over.

SL should have a requirement sure, so should FC, Captain is probably a bit much (I'd argue FC has alot more to do than Captain). Putting a timelock on CL is just ridiculous, its a RP role. I think your really underestimating new players here.

**Comment by terranaut1** [comment]
i commented this on the discord as i was pinged for my opinion but it should probably be here too
a timelock on command roles is fine to me, but on crew ranks (SL and below, maybe an hour as any crew rank for SL, so essentially 1 round) seems a bit silly to me
theres no reason someone shouldnt be able to play engineer or medic

**Comment by lKiseki** [comment]
Unironically, I was one of the people that suggested a much more increased timelock for roles such as captain and xeno queen, particularly because there's a minor tendency with such roles where new players will take them and, lacking knowledge and experience, will proceed to do many notable mistakes that end up perjudicating their respective teammates. While I can understand why crew roles could use lowered timelocks, I would personally prefer it if roles such as the aforementioned ones had drastically increased timelocks, somewhere around thirty to five hours of playtime total (not necessarily limited to marine/xeno respectively).

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -111,8 +111,10 @@ MEDICAL_OFFICER, MEDICAL_RESEARCHER, SQUAD_LEADER, SQUAD_SPECIALIST, SQUAD_SMART
 #define CRYO_DELTA		"Delta Squad"
 
 
-#define XP_REQ_INTERMEDIATE 60
-#define XP_REQ_EXPERIENCED 180
+#define XP_REQ_UNSEASONED 60
+#define XP_REQ_INTERMEDIATE 180
+#define	XP_REQ_EXPERIENCED 600
+#define	XP_REQ_EXPERT 900
```
Comment (SplinterGP) [inline code/__DEFINES/jobs.dm:117]: too much space ?

Comment (BraveMole) [reply code/__DEFINES/jobs.dm:117]: It looks fine on VSC, weird

**Comment by davidchan13** [comment]
> Additionally, although I'm not sure if it does, the timelock for xeno queen should also apply when attempting to evolve from Drone/Shrike to Queen and restrict them from doing so.

This is bad, if no Queen or Shrike is present during distress for an allotted time it's an immediate forfeit ala hivemind collapse. You're literally suggesting that xenos should lose the game because they didn't have a living player eligible, or that during a hivemind/comms blackout  the obvious drone choice to be new ruler couldn't communicate that the game actively was blocking them from evolving due to time restrictions.

**Comment by BraveMole** [comment]
> > Additionally, although I'm not sure if it does, the timelock for xeno queen should also apply when attempting to evolve from Drone/Shrike to Queen and restrict them from doing so.
> 
> This is bad, if no Queen or Shrike is present during distress for an allotted time it's an immediate forfeit ala hivemind collapse. You're literally suggesting that xenos should lose the game because they didn't have a living player eligible, or that during a hivemind/comms blackout the obvious drone choice to be new ruler couldn't communicate that the game actively was blocking them from evolving due to time restrictions.

You're mistaken, he just want to prevent for going queen, not shrike

**Review by TiviPlus — APPROVED** [review]
Admin say ok

**Pull request merged**